### PR TITLE
DOC: Start documentation building with sphinx

### PR DESCRIPTION
**Pull request by oesteban**
- [x] Uses autodoc to generate the API of the library.
- [x] Revised most of the docstrings, which will need to be improved as we advance.
- [x] Homogenized parameters formatting to use numpydoc format.
- [x] Uses sphinxcontrib-versioning to store docs on gh-pages.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+sys.path.insert(0, str(Path(__file__).parent.parent.absolute()))
+sys.path.insert(1, str(Path(__file__).parent / 'sphinxext'))
+from pydra import __version__
+from github_link import make_linkcode_resolve
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'Pydra: A simple dataflow engine with scalable semantics'
+copyright = '2020, The Nipype Developers team'
```
Comment (satra) [inline docs/conf.py]:
```suggestion
copyright = '2019 - 2020, The Nipype Developers team'
```

**Comment by satra** [comment]
thanks so much @oesteban - you are in doc mode!!

can we add executed versions of the pydra tutorial notebooks to the docs during the doc build process? this will also help us make things consistent.

**Comment by oesteban** [comment]
There should be an extension to do that easily. I'll look that up.

I thought helping with documentation was a great way of diving into the code :). Outstanding work there @djarecka and @mgxd !

**Comment by oesteban** [comment]
Okay, I'm going to go ahead and merge because I don't want to leave the CircleCI deployment untested. We can always work from this PR.

**Pull request merged**